backend/s3/s3.go

```diff
@@ -2958,9 +2967,32 @@ func (o *Object) Storable() bool {
 	return true
 }
 
+func (o *Object) downloadFromURL(ctx context.Context, bucketPath string, options ...fs.OpenOption) (in io.ReadCloser, err error) {
+	url := o.fs.opt.DownloadURL + bucketPath
+	var resp *http.Response
+	opts := rest.Opts{
+		Method:  "GET",
+		RootURL: url,
+		Options: options,
+	}
+	err = o.fs.pacer.Call(func() (bool, error) {
+		resp, err = o.fs.srvRest.Call(ctx, &opts)
+		return o.fs.shouldRetry(ctx, err)
+	})
+	if err != nil {
+		return nil, err
+	}
+	return resp.Body, err
```

Comment:
This should be setting the metadata as is done here if possible.

https://github.com/rclone/rclone/blob/bb0c4ad2d8af8af1ff9abea54dbdf97eadc9018f/backend/s3/s3.go#L3013-L3029

I don't know whether the metadata is returned when objects are fetched via the CDN - I need your advice there!

Reply:
Thank you for the review.

I just checked out the response headers from CDN. It does contain the metadata. Will update the PR.

```
HTTP/1.1 200 OK
Content-Type: image/jpeg
Content-Length: 3968662
Connection: keep-alive
Date: Sat, 02 Oct 2021 10:57:28 GMT
Access-Control-Allow-Origin: *
Access-Control-Allow-Methods: GET, PUT, POST, DELETE, HEAD
Access-Control-Expose-Headers: ETag
Last-Modified: Sat, 18 Apr 2020 16:41:17 GMT
ETag: "08bfadc133h323f6663f434ecc9743"
x-amz-storage-class: STANDARD_IA
x-amz-version-id: null
Accept-Ranges: bytes
Server: AmazonS3
X-Edge-Origin-Shield-Skipped: 0
Vary: Access-Control-Request-Headers,Access-Control-Request-Method
X-Cache: Miss from cloudfront
Via: 1.1 dsfefefewefwfw.cloudfront.net (CloudFront)
X-Amz-Cf-Pop: MAA50-C1
X-Amz-Cf-Id: fdfdfsd_wWfsfsfksksks==
```